include/yateto/TensorView.h

```diff
@@ -132,22 +132,29 @@ namespace yateto {
     }
 
     template<typename Head>
-    void isInRange(uint_t start[Dim], uint_t stop[Dim], int dim, Head head) {
-      assert(static_cast<uint_t>(head) >= start[dim]);
-      assert(static_cast<uint_t>(head) < stop[dim]);
+    bool isInRange(uint_t start[Dim], uint_t stop[Dim], int dim, Head head) {
+      return static_cast<uint_t>(head) >= start[dim] && static_cast<uint_t>(head) < stop[dim];
     }
 
     template<typename Head, typename... Tail>
-    void isInRange(uint_t start[Dim], uint_t stop[Dim], int dim, Head head, Tail... tail) {
-      assert(static_cast<uint_t>(head) >= start[dim]);
-      assert(static_cast<uint_t>(head) < stop[dim]);
-      isInRange(start, stop, dim+1, tail...);
+    bool isInRange(uint_t start[Dim], uint_t stop[Dim], int dim, Head head, Tail... tail) {
+      return static_cast<uint_t>(head) >= start[dim]
+             && static_cast<uint_t>(head) < stop[dim]
+             && isInRange(start, stop, dim+1, tail...);
     }
-  
+
+    template<typename... Entry>
+    bool isInRange(Entry... entry) {
+      static_assert(sizeof...(entry) == Dim,
+                  "Number of arguments to isInRange(...) does not match Tensor's dimension.");
```

Comment:
suggestion for rewrite `"[...] does not match the dimension of the tensor"`. (iirc `'s` was mostly used for animate objects?)

Reply:
Eh, it's not incorrect here IMO:
https://english.stackexchange.com/a/30402
"An important exception is the so called thematic genitive: if a noun has gained strong [topical](https://en.wikipedia.org/wiki/Topic_and_comment) value, because it is central in a discussion or description, it may get the possessive s."

(I personally would also use of, but I used phrasing that is consistent with other static_asserts in the same file and see no need to change it)

Reply:
Yup, but is it really the topic here? Is that really obvious enough here? (I would fully agree if there was a previous sentence saying something like "The tensor has dimension N." or so) But ok, that's debatable. All in all, a storm in a teapot.

... maybe it would work as well to just write `tensor dimension`, without `'s`. :D